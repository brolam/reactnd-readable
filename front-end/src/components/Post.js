import React from 'react'
import PropTypes from 'prop-types'
import Moment from 'moment';
import VoteScore from './VoteScore'

function Post(props) {
  const { post } = props
  return (
    <div className="post" onClick={props.onSelected}>
      <div className="post-header">
        <div className="post-header-title">
          <span className={"category " + post.category} >{post.category[0].toUpperCase() + post.category.slice(1)}</span> {post.title}
        </div>
        <div className="post-header-published">
          <span className="published">{Moment(post.timestamp).from(new Date())}</span>
        </div>
      </div>
      <div className="post-header-author">by {post.author}</div>
      <div className="post-body">{post.body}</div>
      <div className="post-footer">
        <VoteScore voteScore={post.voteScore} />
      </div>
    </div>
  )
}

Post.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    voteScore: PropTypes.number.isRequired,
  }),
  onSelected: PropTypes.func.isRequired
}

export default Post