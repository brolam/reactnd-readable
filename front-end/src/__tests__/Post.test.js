import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Post from '../components/Post'

test('render without error', () => {
  ReactDOM.render(
    <Post post={postRedux} comments={postReduxComments} />,
    document.createElement('div'));
})

test('last Snapshot', () => {
  const post = renderer.create(<Post post={postRedux} comments={postReduxComments} />);
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - Edit Post', () => {
  const post = renderer.create(<Post
    post={postRedux}
    comments={postReduxComments}
    isEditPost={true} />);
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - New Comment', () => {
  const post = renderer.create(<Post
    post={postRedux}
    comments={postReduxComments}
    isNewComment={true}
  />);
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - Edit Comment', () => {
  const post = renderer.create(<Post
    post={postRedux}
    comments={postReduxComments}
    isEditComment={true}
    selectedComment={postReduxComments[0]}
  />);
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - show question delete comment', () => {
  const post = renderer.create(<Post
    post={postRedux}
    comments={postReduxComments}
    isShowQuestionDelComment={true}
    selectedComment={postReduxComments[0]}
  />);
  let tree = post.toJSON();
  expect(tree).toMatchSnapshot();
})

const postRedux = {
  category: "redux",
  id: "7xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn Redux",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}

const postReduxComments = [
  {
    id: '894tuq4ut84ut8v4t8wun89g',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1468166872634,
    body: 'Hi there! I am a COMMENT.',
    author: 'thingtwo',
    voteScore: 6,
    deleted: false,
    parentDeleted: false
  },
  {
    id: '8tu4bsun805n8un48ve89',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1469479767190,
    body: 'Comments. Are. Cool.',
    author: 'thingone',
    voteScore: -5,
    deleted: false,
    parentDeleted: false
  }
]


