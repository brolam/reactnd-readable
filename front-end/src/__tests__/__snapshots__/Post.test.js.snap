// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
<div
  className="app"
>
  <div
    className="post-page-header"
  >
    <a
      className="post-page-header-back"
      onClick={[Function]}
    >
      Back
    </a>
    <div
      className="post-page-header-title"
    >
      <span
        className="redux"
      >
        redux
      </span>
      <span>
        Udacity is the best place to learn Redux
      </span>
    </div>
    <div
      className="post-page-header-buttons"
    >
      <button
        className="edit-button"
        href="/"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="liked"
        onClick={undefined}
      >
        Liked 
        <span>
          5
        </span>
      </button>
      <button
        className="not-liked"
        onClick={undefined}
      >
        Not like
      </button>
    </div>
  </div>
  <div
    className="post-content"
  >
    <div
      className="post-page-body"
    >
      <div
        className="post-page-author"
      >
        by 
        thingtwo
         
        a year ago
      </div>
      Everyone says so after all.
    </div>
    <div
      className="post-comments-title"
    >
      <span>
        Commentes(
        2
        )
      </span>
      <div
        className="order-options"
      >
        <select>
          <option
            value="voteScore"
          >
            Vote Score
          </option>
          <option
            value="publishedDate"
          >
            Published date
          </option>
        </select>
      </div>
    </div>
    <div
      className="post-page-comments"
    >
      <div
        className="post-comment"
      >
        <div
          className="post-comment-author"
        >
          by 
          thingtwo
           
          a year ago
        </div>
        <div>
          Hi there! I am a COMMENT.
        </div>
        <div
          className="post-comment-footer"
        >
          <button
            className="edit-button"
            href="/"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="liked"
            onClick={undefined}
          >
            Liked 
            <span>
              6
            </span>
          </button>
          <button
            className="not-liked"
            onClick={undefined}
          >
            Not like
          </button>
        </div>
      </div>
      <div
        className="post-comment"
      >
        <div
          className="post-comment-author"
        >
          by 
          thingone
           
          a year ago
        </div>
        <div>
          Comments. Are. Cool.
        </div>
        <div
          className="post-comment-footer"
        >
          <button
            className="edit-button"
            href="/"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="liked"
            onClick={undefined}
          >
            Liked 
            <span>
              -5
            </span>
          </button>
          <button
            className="not-liked"
            onClick={undefined}
          >
            Not like
          </button>
        </div>
      </div>
    </div>
    <div
      className="flat-button"
      onClick={[Function]}
    >
      <a
        className="add redux"
      >
        Add Comment
      </a>
    </div>
  </div>
</div>
`;
